src/USER-PHONON/dynamical_matrix.cpp

```diff
@@ -0,0 +1,491 @@
+                            //dynmat has length dynlen
+                            //dynlen = (group->count(igroup))*3;
+                            //currently dynmat is being called to natoms*3
+                            //Also with this implementation
+                            //I am not recovering the correct dynamical matrix
+                            //if I have more than 1 core
+                            dynmat[(i-1)*3+alpha][(j-1)*3+beta] -= -f[j-1][beta];
+                            dynmat[(i-1)*3+alpha][(j-1)*3+beta] /= (2 * del * imass);
+                            dynmat[(i-1)*3+alpha][(j-1)*3+beta] *= conversion;
+                        }
+                    }
+                }
+                displace_atom(local_idx,alpha,1);
+            }
+        }
+    }
+
+
+    memory->create(final_dynmat,int(dynlen),int(dynlen),"dynamic_matrix_buffer:buf");
+    for (int i = 0; i < dynlen; i++)
+        MPI_Reduce(dynmat[i], final_dynmat[i], int(dynlen), MPI_DOUBLE, MPI_SUM, 0, world);
+
+    if (screen && me ==0 ) fprintf(screen,"Finished Calculating Dynamical Matrix\n");
+}
+
+/* ----------------------------------------------------------------------
+   write dynamical matrix
+------------------------------------------------------------------------- */
+
+void DynamicalMatrix::writeMatrix()
+{
+    // print file comment lines
+    if (!binaryflag && fp) {
+        clearerr(fp);
+        for (int i = 0; i < dynlen; i++) {
+            for (int j = 0; j < dynlen; j++) {
+                if ((j+1)%3==0) fprintf(fp, "%4.8f\n", final_dynmat[j][i]);
+                else fprintf(fp, "%4.8f ",final_dynmat[j][i]);
+            }
+        }
+    }
+    if (ferror(fp))
+        error->one(FLERR,"Error writing to file");
+
+    if (binaryflag && fp) {
+        clearerr(fp);
+        fwrite(&final_dynmat[0], sizeof(double), dynlen * dynlen, fp);
+        if (ferror(fp))
+            error->one(FLERR, "Error writing to binary file");
+    }
+}
+
+/* ----------------------------------------------------------------------
+  Displace atoms
+   ---------------------------------------------------------------------- */
+
+void DynamicalMatrix::displace_atom(int local_idx, int direction, int magnitude)
+{
+    double **x = atom->x;
+    int *sametag = atom->sametag;
+    int j = local_idx;
+
+    x[local_idx][direction] += del*magnitude;
+
+    while (sametag[j] >= 0){
+        j = sametag[j];
+        x[j][direction] += del*magnitude;
+    }
+}
+
+
+/* ----------------------------------------------------------------------
+   evaluate potential energy and forces
+   may migrate atoms due to reneighboring
+   return new energy, which should include nextra_global dof
+   return negative gradient stored in atom->f
+   return negative gradient for nextra_global dof in fextra
+------------------------------------------------------------------------- */
+
+void DynamicalMatrix::energy_force(int resetflag)
```

Comment:
calling this function `energy_force()` is a misnomer, since you are setting `eflag` and `vflag` to zero, and thus do not tally energy or virial contributions.

Reply:
I just wanted to make it look similar to other class methods. Would you have me change other classes?

Reply:
Since this is an "internal" function, its name only has to make sense within this class.